src/transformers/modeling_bert.py

```diff
@@ -957,7 +957,29 @@ def forward(
             ltr_lm_loss = loss_fct(prediction_scores.view(-1, self.config.vocab_size), lm_labels.view(-1))
             outputs = (ltr_lm_loss,) + outputs
 
-        return outputs  # (masked_lm_loss), (ltr_lm_loss), prediction_scores, (hidden_states), (attentions)
+        return outputs  # (ltr_lm_loss), (masked_lm_loss), prediction_scores, (hidden_states), (attentions)
+
+    def prepare_inputs_for_generation(self, input_ids, attention_mask=None, **model_kwargs):
```

Comment:
Since Bert is usually pretrained with masked language modeling, for generation a new "dummy" token is added, on which we put a mask so that the next token can be predicted by Bert. This is similar to how it is done for XLNet e.g.: https://github.com/huggingface/transformers/blob/857ccdb259b7e46c60cf86c58b7ab038c63e4d4e/src/transformers/modeling_xlnet.py#L941